build.js

```diff
@@ -16,81 +16,168 @@ function getDirname () {
+	// console.log('Preparing folder', folderPath)
+	// create if it doesn't exists yet
+	if (!fs.existsSync(folderPath)) {
+		fs.mkdirSync(folderPath)
+		console.log('Created folder', folderPath)
+	}
+}
+
+// creates a dictionary of (filename: sourceFile) for Handlebars
+function makeComponentDictionary(globPattern) {
+	console.log('Looking for components matching: ' + globPattern)
+
+	const componentSourceFiles = glob.sync(globPattern)
+	const components = {}
+	for (const componentSourceFile of componentSourceFiles) {
+		const componentName = componentSourceFile
+			.split('/').pop() // get the actual filename
+			.replace(/\.html$/, '') // remove .html at the end
+	
+		console.log('Found component', componentName)
+
+		// parse component
+		const fileString = fs.readFileSync(componentSourceFile, "utf8")
+		if (fileString.indexOf('TEMPORARY_OPENING_SIGNATURE ') == 0) {
+			console.log(componentName, 'was identified as a compound component')
+
+			// split by sections
+			let sections = fileString.split('\r\nTEMPORARY_OPENING_SIGNATURE ')
+			sections[0] = sections[0].replace('TEMPORARY_OPENING_SIGNATURE ', '')
+			// seperate sections into name and content
+			sections = sections.map(section => section.split(' TEMPORARY_CLOSING_SIGNATURE\r\n'))
+			// add component sections to dictionary
+			let componentDict = {}
+			sections.map(section => componentDict[section[0]] = new Handlebars.SafeString(section[1]))
+			// add compound component to main component dictionary
+			components[componentName] = componentDict
+		}
+		else {
+			// Handlebars escapes the '<' and '>' characters by default, so we need to
+			// explain that the strings are safe
+			// API reference: http://handlebarsjs.com/reference.html
+			components[componentName] = new Handlebars.SafeString(fileString)
+		}
+	
+		
+	}
+	return components
+}
+
+function transpileUsingNestedHandlebars(fileString, fileToCreate, components) {
+	// this is where we transpile!
+	// https://github.com/wycats/handlebars.js/#usage
+	let template
+
+	// repeated Handlebar replacement to allow for nesting
+	while (fileString.indexOf('{{') !== -1) {
+		template = Handlebars.compile(fileString)
+		fileString = template(components)
+	}
+	// write output to file
+	fs.writeFileSync(fileToCreate, fileString)
+	console.log('Transpiled file', fileToCreate)
 }
 
+// create the target build/ folder if it doesn't exist yet
+makeFolder('./build/')
+// create base html folders (en/sk versions)
+makeFolder(getDirname() + '/build/html/')
+makeFolder(getDirname() + '/build/html/en/')
+makeFolder(getDirname() + '/build/html/sk/')
+
 // first we prepare all folders in the build folder
 const sourceFolders = glob.sync(getDirname() + '/source/**/', {})
 for (const sourceFolder of sourceFolders) {
 	// identifies the new folder to be created
-	const folderToCreate = getBuildFilePathFromSourceFilePath(sourceFolder)
 
-	console.log('Preparing folder ', folderToCreate)
+	// checks if it is in the html folder
```

Comment:
what does "it" refer to here?

Reply:
It is the source folder currently being copied